web/pdf_print_service.js

```diff
@@ -200,7 +211,17 @@
       if (!activeService) {
         console.error('Expected print service to be initialized.');
       }
-      activeService.renderPages().then(startPrint, abort);
+      var activeServiceOnEntry = activeService;
+      activeService.renderPages().then(function() {
+        activeServiceOnEntry.throwIfInactive();
+        return startPrint(activeServiceOnEntry);
+      }).catch(function() {
+        // Ignore any error messages.
+      }).then(function() {
+        if (activeServiceOnEntry.active) {
+          abort();
```

Comment:
`activeServiceOnEntry.active` is true if the currently active service is still the same as the service from when the print task started.

If somehow the task was aborted early (e.g. abort was called early), then the "global" activeService is null. Now the interesting case is when another print job is started immediately after abortion of the previous task. Then the `if (activeService)` condition is true, but the async call to abort from the original print request actually cancels the second request instead of the first request (which was already cancelled in the first place).